serializers/shared/src/main/java/io/pravega/schemaregistry/serializers/MultiplexedAndGenericDeserializer.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.serializers;
+
+import com.google.common.base.Preconditions;
+import io.pravega.schemaregistry.client.SchemaRegistryClient;
+import io.pravega.schemaregistry.common.Either;
+import io.pravega.schemaregistry.contract.data.SchemaInfo;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+class MultiplexedAndGenericDeserializer<T, G> extends AbstractDeserializer<Either<T, G>> {
```

Comment:
What are the cases for this deserializer? I see it being used in `jsonTypedOrGenericDeserializer`, and I understand it as the case in which we have a mix of types for which we need to use both provided and a generic deserializer.

Reply:
Yes, basically for event sourcing scenarios where users want to fall back on generic deserializer if they dont have a typed object to deserialize into. 

Its used in all three - avro, protobuf and json